crates/oxc_parser/src/js/declaration.rs

```diff
@@ -42,6 +42,14 @@ impl<'a> Parser<'a> {
         }
     }
 
+    pub(crate) fn parse_using(&mut self) -> Result<Statement<'a>> {
+        let using_decl = self.parse_using_declaration(StatementContext::StatementList)?;
+
+        self.expect(Kind::Semicolon)?;
```

Comment:
try `self.asi()?`? This seems like a asi.

Reply:
I don't think so? but maybe wrong - looks like the grammer expects a semi - it's not optional

`using [no LineTerminator here] [lookahead ≠ await] BindingList[?In, ?Yield, ?Await, ~Pattern] ;`

Reply:
It's an asi. Checkout all the other declarations, or test it with the typescript playgounrd. The spec regarding asi is rather confusing ... https://tc39.es/ecma262/#sec-automatic-semicolon-insertion

Reply:
ah gotcha, thank you